blaze-client/src/main/scala/org/http4s/client/blaze/BlazeClient.scala

```diff
@@ -4,32 +4,17 @@ import org.http4s.blaze.pipeline.Command
 import org.http4s.client.Client
 import org.http4s.{Request, Response}
 
-import scala.concurrent.{ExecutionContext, Future}
+import scala.concurrent.ExecutionContext
 import scala.util.{Failure, Success, Try}
 import scalaz.concurrent.Task
 import scalaz.stream.Process.eval_
 import scalaz.{-\/, \/-}
 
 /** Base on which to implement a BlazeClient */
-trait BlazeClient extends PipelineBuilder with Client {
-
-  implicit protected def ec: ExecutionContext
-
-  /** Recycle or close the connection
-    * Allow for smart reuse or simple closing of a connection after the completion of a request
-    * @param request [[Request]] to connect too
-    * @param stage the [[BlazeClientStage]] which to deal with
-    */
-  protected def recycleClient(request: Request, stage: BlazeClientStage): Unit = stage.shutdown()
-
-  /** Get a connection to the provided address
-    * @param request [[Request]] to connect too
-    * @param fresh if the client should force a new connection
-    * @return a Future with the connected [[BlazeClientStage]] of a blaze pipeline
-    */
-  protected def getClient(request: Request, fresh: Boolean): Future[BlazeClientStage]
-
+final class BlazeClient(manager: ConnectionManager, ec: ExecutionContext) extends Client {
```

Comment:
I would see this taking an implicit ExecutorService.  If you need an ExecutionContext in the bowels, I have a conversion on a branch.